src/components/QrcodeStream.vue

```diff
@@ -6,279 +6,282 @@
     The first element is at the very bottom and subsequent elements are added on top.
     -->
     <video
-      ref="video"
+      ref="videoRef"
       :class="{ 'qrcode-stream-camera--hidden': !shouldScan }"
       class="qrcode-stream-camera"
       autoplay
       muted
       playsinline
-    ></video>
+    />
 
-    <canvas
-      ref="pauseFrame"
-      v-show="!shouldScan"
-      class="qrcode-stream-camera"
-    ></canvas>
+    <canvas ref="pauseFrameRef" v-show="!shouldScan" class="qrcode-stream-camera" />
 
-    <canvas ref="trackingLayer" class="qrcode-stream-overlay"></canvas>
+    <canvas ref="trackingLayerRef" class="qrcode-stream-overlay" />
 
     <div class="qrcode-stream-overlay">
-      <slot></slot>
+      <slot />
     </div>
   </div>
 </template>
 
-<script>
-import { keepScanning } from "../misc/scanner.js";
-import Camera from "../misc/camera.js";
-import CommonAPI from "../mixins/CommonAPI.vue";
-
-export default {
-  name: "qrcode-stream",
-
-  mixins: [CommonAPI],
-
-  props: {
-    camera: {
-      type: String,
-      default: "auto",
+<script setup lang="ts">
+import type { DetectedBarcode } from '@sec-ant/barcode-detector'
+import { onBeforeUnmount, computed, onMounted, ref, watch } from 'vue'
+
+import { adaptOldFormat, keepScanning } from '../misc/scanner'
+import Camera from '../misc/camera'
+import { useCommonApi } from '../composables/useCommonApi'
+import type { Point } from '../types/types'
+
+const props = defineProps({
+  camera: {
+    type: String,
+    default: 'auto',
+    validator(camera: string) {
+      return ['auto', 'rear', 'front', 'off'].includes(camera)
+    }
+  },
+  torch: {
+    type: Boolean,
+    default: false
+  },
+  track: {
+    type: Function
+  }
+})
+
+const emit = defineEmits(['detect', 'decode', 'init'])
+const { onDetect } = useCommonApi(emit)
+
+// refs
+const pauseFrameRef = ref<HTMLCanvasElement>()
+const trackingLayerRef = ref<HTMLCanvasElement>()
+const videoRef = ref<HTMLVideoElement>()
+
+// data
+const cameraInstance = ref<Awaited<ReturnType<typeof Camera>>>()
+const isMounted = ref(false)
+
+// computations
+const shouldScan = computed(
+  () =>
+    isMounted.value &&
+    props.camera !== 'off' &&
+    cameraInstance.value &&
+    videoRef.value &&
+    videoRef.value.readyState > 1
```

Comment:
I think technically we don't have to check for 
```
videoRef.value &&
videoRef.value.readyState > 1
```
here but correct me if I'm wrong. `videoRef.value` is truthy iff `isMounted.value` is `true` and `videoRef.value.readyState > 1` holds iff `cameraInstance.value` is truthy. The latter holds because we [await the `loadeddata` event](https://github.com/gruhn/vue-qrcode-reader/blob/edf2635c2e424d3bb443776c9f1fbc6e3440d6d3/src/misc/camera.ts#L135) when we initialize `cameraInstance`. Also, I'm not sure about `videoRef.value` but I guess at least `videoRef.value.readyState` is not reactive so Vue can't detect when this value changes, right?

Reply:
According to my testing, you are right!